nva-datamodel-java/src/test/java/no/unit/nva/model/file/FileModelTest.java

```diff
@@ -172,6 +170,48 @@ void shouldNotBeVisibleForNonOwnerWhenUnpublished() throws JsonProcessingExcepti
         assertThat(unmapped.isVisibleForNonOwner(), equalTo(false));
     }
 
+    @Test
+    void objectMapperShouldSerializeAndDeserializePublishedVersion() throws JsonProcessingException {
+        var unpublishedFile = new UnpublishedFile(UUID.randomUUID(), randomString(), randomString(), 10L, null,
+                                                  false, false, null, null, randomString());
+        var unpublishedFileString = unpublishedFile.toString();
+        var publicationAfterRoundTrip = dataModelObjectMapper.readValue(unpublishedFileString, UnpublishedFile.class);
+        assertThat(publicationAfterRoundTrip.getPublisherVersion(), is(equalTo(PublisherVersion.ACCEPTED_VERSION)));
+    }
+
+    @Test
+    void objectMapperShouldSerializeAndDeserializePublisherVersionFromBoolean() throws JsonProcessingException {
+        var unpublishedFile = new UnpublishedFile(UUID.randomUUID(), randomString(), randomString(), 10L, null,
+                                                  false, true, null, null, randomString());
+        var unpublishedFileString = unpublishedFile.toString();
+        var unpublishedFileAfterRoundtrip = dataModelObjectMapper.readValue(unpublishedFileString,
+                                                                            UnpublishedFile.class);
+        assertThat(unpublishedFileAfterRoundtrip.getPublisherVersion(),
+                   is(equalTo(PublisherVersion.PUBLISHED_VERSION)));
+    }
+
+    @Test
+    void objectMapperShouldSerializePublisherVersionFromEnum() throws JsonProcessingException {
+        var unpublishedFile = new UnpublishedFile(UUID.randomUUID(), randomString(), randomString(), 10L, null,
+                                                  false, PublisherVersion.PUBLISHED_VERSION, null, null,
+                                                  randomString());
+        var unpublishedFileString = unpublishedFile.toString();
+        var unpublishedFileAfterRoundTrip = dataModelObjectMapper.readValue(unpublishedFileString,
+                                                                            UnpublishedFile.class);
+        assertThat(unpublishedFileAfterRoundTrip.getPublisherVersion(),
+                   is(equalTo(PublisherVersion.PUBLISHED_VERSION)));
+    }
+
+    @Test
+    void publisherVersionIsSetToNullIfNullIsRoundTrippedToNull() throws JsonProcessingException {
+        var unpublishedFile = new UnpublishedFile(UUID.randomUUID(), randomString(), randomString(), 10L, null,
```

Comment:
Maybe we should not accept null, but instead not applicable?

publisherVersion is only applicable to certain publicationInstances as far as I am aware.

Reply:
If it may be null…? If not, I guess "Unknown"?